gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -528,10 +528,9 @@ private int readObjectContentFromGCS(ByteBuffer byteBuffer) throws IOException {
       int skipBytes = Math.toIntExact(positionForNextRead - positionInGrpcStream);
       if (skipBytes >= 0 && skipBytes < content.size()) {
         content = content.substring(skipBytes);
-        positionInGrpcStream = positionForNextRead;
+        positionInGrpcStream += skipBytes;
       } else if (skipBytes >= content.size()) {
         positionInGrpcStream += content.size();
-        positionForNextRead = positionInGrpcStream;
```

Comment:
it wont hurt if we keep this correct?

Reply:
It will - that  would be wrong. positionForNextRead could be higher than positionInGrpcStream